Source/common/SNTFileInfo.m

```diff
@@ -712,17 +713,15 @@ - (NSString *)resolvePath:(NSString *)path bundle:(NSBundle **)bundle {
 
 ///
 ///  Cache and return a MOLCodeSignChecker for the given file.  If there was an error creating the
-///  code sign checker, the error may be obtained from the read-only codesignCheckerError property.
+///  code sign checker it will be returned in the passed-in error parameter.
 ///
-- (MOLCodesignChecker *)codesignChecker {
-  @synchronized (self) {
-    if (!self.cachedCodesignChecker) {
-      NSError *error;
-      self.cachedCodesignChecker = [[MOLCodesignChecker alloc]
-          initWithBinaryPath:self.path error:&error];
-      _codesignCheckerError = error;
-    }
+- (MOLCodesignChecker *)codesignCheckerWithError:(NSError **)error {
+  if (!self.cachedCodesignChecker) {
```

Comment:
Added a check against the existence of self.codesignCheckerError so that if a previous call returned an error, it won't attempt to create the codesign checker again.  MOLCodesignChecker's initWithBinaryPath:error:  method seems to return a non-nil object even when the file is not codesigned, so this change addresses the case where there was a caught error and init returned nil.  If init returns nil without setting an error, then I think I'm OK with trying to create the codesign checker again.